### ISSUE TITLE
introduce_parameter SUCCESS leaves call sites missing new required argument → CS7036 (FindReferences on pre-rewrite symbol after signature change)

### ISSUE THREAD

**Issue by JoshuaRamirez** [JoshuaRamirez/RoslynMcpServer#1266]
## Summary
`introduce_parameter` returns SUCCESS but leaves existing call sites without the new required argument → **CS7036**. Documented behavior is to update call sites when promoting a local with an initializer.

## Repro (master af649e14a312ecd48686210022721e43ee7cf076)
```csharp
using System;

public class Greeter
{
    public void Speak(string name)
    {
        string greeting = "hello";
        Console.WriteLine(greeting + name);
    }
}

public static class Program
{
    public static void Main()
    {
        new Greeter().Speak("world");
    }
}
```
Run introduce_parameter on local `greeting`.

### After (SUCCESS)
```csharp
public void Speak(string name, string greeting)
{
    Console.WriteLine(greeting + name);
}
// ...
new Greeter().Speak("world"); // missing greeting → CS7036
```

Also WRONG with `var greeting = "hi " + name;` and with uninitialized `string greeting;` + later assign (initializer null → call-site update skipped entirely, still SUCCESS + CS7036).

## Expected
Update each call site with the initializer expression as the new argument, **or** fail clearly if callers cannot be updated. Do not report SUCCESS leaving CS7036.

## Root cause
`src/RoslynMcp.Core/Refactoring/Extract/IntroduceParameterOperation.cs` ~L175–207:
- Method signature is rewritten first on `newSolution`.
- Call-site pass uses `SymbolFinder.FindReferencesAsync(methodSymbol, newSolution, …)` with `methodSymbol` from the **pre-rewrite** model. After the only declaration’s signature changes, references are typically empty → callers untouched.
- Still returns `RefactoringResult.Succeeded`.
- Call-site pass is also gated on `initializer != null` (~L183).

Unit tests in `IntroduceParameterOperationTests` exercise methods with **no call sites**, so this never fails CI.

## Related
- #1263 introduce_parameter × partial (parts diverge; callers may update via untouched other part)
- #528 UnsupportedCallSite *validation* skip (different — not this call-site miss)

## WRONG=yes
Live probe: `/workspace/fd-make-static-intro-probe/out.txt` (ordinary-with-init-*).
